data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/CsvLogstashPluginAttributesMapper.java

```diff
@@ -0,0 +1,113 @@
+    protected static final String DATA_PREPPER_COLUMN_NAMES_SOURCE_KEY = "column_names_source_key";
+    protected static final String DATA_PREPPER_COLUMN_NAMES = "column_names";
+    protected static final String LOGSTASH_AUTOGENERATE_COLUMN_NAMES_ATTRIBUTE_NAME = "autogenerate_column_names";
+
+    protected static final String LOGSTASH_SKIP_EMPTY_ROWS_ATTRIBUTE_NAME = "skip_empty_rows";
+    protected static final String LOGSTASH_SKIP_EMPTY_COLUMNS_ATTRIBUTE_NAME = "skip_empty_columns";
+    protected static final String LOGSTASH_SKIP_HEADER_ATTRIBUTE_NAME = "skip_header";
+    protected static final String LOGSTASH_CONVERT_ATTRIBUTE_NAME = "convert";
+    protected static final String LOGSTASH_AUTODETECT_COLUMN_NAMES_EXCEPTION_MESSAGE = "Data Prepper logstash conversion does not " +
+            "support the autodetect_column_names setting in Logstash configuration. Consider manually specifying a Data Prepper config " +
+            "file with a CSV Processor for header autodetection functionality.";
+    protected static final String LOGSTASH_SKIP_EMPTY_ROWS_EXCEPTION_MESSAGE = "Data Prepper logstash conversion does not support the " +
+            "skip_empty_columns setting in Logstash configuration. Consider using a delete_entries processor with the empty columns you " +
+            "wish to drop.";
+    protected static final String LOGSTASH_SKIP_EMPTY_COLUMNS_EXCEPTION_MESSAGE = "Data Prepper logstash conversion does not support the " +
+            "skip_empty_rows setting in Logstash configuration. Consider using a drop_events processor to drop empty rows.";
+    protected static final String LOGSTASH_SKIP_HEADER_EXCEPTION_MESSAGE = "Data Prepper logstash conversion does not support the " +
+            "skip_header setting in Logstash configuration. Consider using a drop_events processor to drop tbe first row.";
+    protected static final String LOGSTASH_CONVERT_EXCEPTION_MESSAGE = "Data Prepper logstash conversion does not support the convert " +
+            "setting in Logstash configuration. This feature may be supported in the future; a separate processor to typecast Event " +
+            "fields is in the Data Prepper roadmap.";
+    @Override
+    protected void mapCustomAttributes(final List<LogstashAttribute> logstashAttributes,
+                                       final LogstashAttributesMappings logstashAttributesMappings,
+                                       final Map<String, Object> pluginSettings) {
+        final Optional<LogstashAttribute> autogenerateColumnNamesAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_AUTOGENERATE_COLUMN_NAMES_ATTRIBUTE_NAME);
+
+        if (logstashAttributeExistsAndIsTrue(autogenerateColumnNamesAttribute)) {
+            pluginSettings.put(
+                    DATA_PREPPER_COLUMN_NAMES,
+                    new ArrayList<String>()
+            );
+        }
```

Comment:
What should be expected behavior if Logstash configuration has columns defined and autogenerate_column_names is set to true?
```
columns => ["a", "b"]
autogenerate_column_names => true
```

This mapper will override the columns field and set it to empty list.
```
column_names: []
```

Reply:
I tested Logstash with both of those settings and it uses the names in `columns` instead of autogenerating, even if `autogenerate_column_names` is `true`. I'll update the `mapCustomAttributes` class to account for this edge case.